cxx/hirm.hh

```diff
@@ -537,11 +542,16 @@ class Relation {
   Relation(const Relation&) = delete;
 };
 
+using RelationVariant =
+    std::variant<Relation<BetaBernoulli>*, Relation<Bigram>*,
+                 // Relation<double, DirichletCategorical>*,
```

Comment:
What's wrong with DirichletCategorical?

Reply:
@emilyfertig interested to hear what you think are the plausible options for parametrized distributions such as this---I feel like it might be a recurrent theme, for instance if we want to generalize the bigram model to n-grams. Including the parameter in the distribution name might get unwieldy quickly---if arguments are real-valued, if are multiple arguments etc.

Reply:
What do you think of allowing something like `categorical has feature animal num_categories=3`, where each line of the schema can have appended some number of strings containing `=` that have to match certain allowed distribution parameters? (That's not a great categorical example, maybe the animal has the feature in the wild, in captivity, or not at all). If we want to pass other args besides distribution parameters, we might have to make bigger changes to the schema, but maybe that's an ok place to start. Interested to hear your thoughts, and if there are alternatives you think we should consider.

Reply:
We could also just use constructor-like syntax:

`categorical(3) has feature animal`